### ISSUE TITLE
Trying to get validator to work.

### ISSUE THREAD

**Issue by dakamojo** [gtgalone/prompt_dialog#12]
Thanks for the nice clean widget.  I'm trying to get the validator to work.  It may be that the validator message is displaying but the dialog still closes immediately.

So I tried the sample custom_message.dart.

First, the sample has some minor issues.  The prompt widget needs a context that is a descendent of the MaterialApp.  So I wrapped the Center Widget with a Builder for simplicity.  Second I got some error related to the obscureText so I commented that out.  Here is my updated sample.

```
import 'package:flutter/material.dart';

import 'package:prompt_dialog/prompt_dialog.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Welcome to Flutter',
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Welcome to Flutter'),
        ),
        body: Builder(builder: (context) {
          return Center(
            child: TextButton(
              onPressed: () async {
                return print(await prompt(
                  context,
                  title: const Text('Would you like to remove?'),
                  initialValue: 'Sure',
                  isSelectedInitialValue: false,
                  textOK: const Text('Yes'),
                  textCancel: const Text('No'),
                  hintText: 'Please write reason',
                  validator: (String? value) {
                    if (value == null || value.isEmpty) {
                      return 'Please enter some text';
                    }
                    return null;
                  },
                  minLines: 2,
                  maxLines: 3,
                  autoFocus: false,
                  // obscureText: true,
                  obscuringCharacter: '*',
                  showPasswordIcon: true,
                  barrierDismissible: true,
                  textCapitalization: TextCapitalization.words,
                ));
              },
              child: const Text('Prompt Dialog'),
            ),
          );
        }),
      ),
    );
  }
}
```

When running this sample the validator message is never presented to the user.

**Comment by dakamojo** [comment]
Update I put a breakpoint and a print statement in the validate itself and neither gets hit so I suspect it's never invoked.  Also at this point, I'm only debugging on web, not mobile.

**Comment by nathanaelhoun** [comment]
> Thanks for the nice clean widget.

+1, this widget is really good!

> Also at this point, I'm only debugging on web, not mobile.

I can confirm the issue also happens on mobile (android)

**Comment by dakamojo** [comment]
It appears that the validate function is never called.  

```
TextButton(
  onPressed: () => Navigator.pop(context, value),
    child: (textOK != null) ? textOK : const Text('OK'),
),
```

should be something like this

```
TextButton(
  onPressed: () {
    if (_formKey.currentState!.validate()) {
      Navigator.pop(context, value);
    },
  child: (textOK != null) ? textOK : const Text('OK'),
),
```

See [https://docs.flutter.dev/cookbook/forms/validation](url)

**Comment by gtgalone** [comment]
@dakamojo Awesome!!

Could you make a pull request?

**Comment by dakamojo** [comment]
@gtgalone I'm making an attempt.  But I'm unsure about how to get the key for the form to call validate.  In the current implementation the key is based on keyCtr which is incremented every time the ShowPasswordIcon is pressed.  Is there a reason the key needs to change?  Or can the key be a GlobalKey that doesn't change?

**Comment by dakamojo** [comment]
I am working on this.  It seems that the key has to be created from a Stateful Widget, a Stateful Builder doesn't work.  I'm refactoring to use a private Stateful Widget but keeping the same functional signature.

**Comment by gtgalone** [comment]
@dakamojo 

Awesome!!

I have just published 1.0.7 with your PR(#13).

Thank you so much!

**Comment by mjablecnik** [comment]
@gtgalone In pub.dev is still version 1.0.6 as latest version..:
![image](https://user-images.githubusercontent.com/2686906/144613317-1a584d96-fb4f-418e-bb83-165dfa4c4bf3.png)

**Comment by gtgalone** [comment]
@mjablecnik Thank you so much.

I forgot it. I am so sorry for that :)

I have just published v1.0.7
